email/bitmessage.go

```diff
@@ -269,22 +219,22 @@ func generateMsgBroadcast(be *Bitmessage, from *identity.Private) (*wire.MsgObje
 	return obj, from.NonceTrialsPerByte, from.ExtraBytes, nil
 }
 
-func generateMsgMsg(be *Bitmessage, from *identity.Private, to *identity.Public) (*wire.MsgObject, uint64, uint64, error) {
+// generateMsg generates a wire.MsgMsg from a Bitmessage.
+func (m *Bitmessage) generateMsg(from *identity.Private, to *identity.Public, expiry time.Duration) (*wire.MsgObject, uint64, uint64, error) {
 	// TODO make a separate function in bmutil that does this.
 	var signingKey, encKey wire.PubKey
 	var destination wire.RipeHash
 	sk := from.SigningKey.PubKey().SerializeUncompressed()[1:]
-	ek := to.EncryptionKey.SerializeUncompressed()[1:]
+	ek := from.EncryptionKey.PubKey().SerializeUncompressed()[1:]
```

Comment:
This is the encryption key to be embedded in the message (of the sender) and not the one actually used for encryption.